src/cascadia/TerminalCore/TerminalApi.cpp

```diff
@@ -137,6 +187,131 @@ bool Terminal::EraseCharacters(const unsigned int numChars)
+        startPos.X = viewport.Left();
+        nlength = viewport.RightInclusive() - startPos.X;
+        break;
+    case DispatchTypes::EraseType::Scrollback:
+        return false;
+    }
+
+    auto eraseIter = OutputCellIterator(L' ', _buffer->GetCurrentAttributes(), nlength);
+    _buffer->Write(eraseIter, startPos);
+    return true;
+}
+
+// Method description:
+// - erases text in the buffer in two ways depending on erase type
+// 1. 'erases' all text visible to the user (i.e. the text in the viewport)
+// 2. erases all the text in the scrollback
+// Arguments:
+// - the erase type
+// Return Value:
+// - true if succeeded, false otherwise
+bool Terminal::EraseInDisplay(const DispatchTypes::EraseType eraseType)
+{
+    // Store the relative cursor position so we can restore it later after we move the viewport
+    const auto cursorPos = _buffer->GetCursor().GetPosition();
+    auto relativeCursor = cursorPos;
+    _mutableViewport.ConvertToOrigin(&relativeCursor);
+
+    // Initialize the new location of the viewport
+    // the top and bottom parameters are determined by the eraseType
+    SMALL_RECT newWin;
+    newWin.Left = _mutableViewport.Left();
+    newWin.Right = _mutableViewport.RightExclusive();
+
+    if (eraseType == DispatchTypes::EraseType::All)
+    {
+        // In this case, we simply move the viewport down, effectively pushing whatever text was on the screen into the scrollback
+        // and thus 'erasing' the text visible to the user
+        const auto coordLastChar = _buffer->GetLastNonSpaceCharacter(_mutableViewport);
+        if (coordLastChar.X == 0 && coordLastChar.Y == 0)
+        {
+            // Nothing to clear, just return
+            return true;
+        }
+
+        short sNewTop = coordLastChar.Y + 1;
+
+        // Increment the circular buffer only if the new location of the viewport would be 'below' the buffer
+        short delta = (sNewTop + _mutableViewport.Height()) - (_buffer->GetSize().Height());
+        for (auto i = 0; i < delta; i++)
+        {
+            _buffer->IncrementCircularBuffer();
+            sNewTop--;
+        }
+
+        newWin.Top = sNewTop;
+        newWin.Bottom = sNewTop + _mutableViewport.Height();
+    }
+    else if (eraseType == DispatchTypes::EraseType::Scrollback)
+    {
+        // We only want to erase the scrollback, and leave everything else on the screen as it is
+        // so we grab the text in the viewport and rotate it up to the top of the buffer
+        COORD scrollFromPos{ 0, 0 };
+        _mutableViewport.ConvertFromOrigin(&scrollFromPos);
+        _buffer->ScrollRows(scrollFromPos.Y, _mutableViewport.Height(), -scrollFromPos.Y);
+
+        // Since we only did a rotation, the text that was in the scrollback is now _below_ where we are going to move the viewport
+        // and we have to make sure we erase that text
+        auto eraseIter = OutputCellIterator(L' ', _buffer->GetCurrentAttributes(), _mutableViewport.RightInclusive());

```

Comment:
another potential UNICODE_SPACE

Reply:
Same issue as above, UNICODE_SPACE is not defined apparently

Reply:
done!